### PR TITLE
Done

### PR DESCRIPTION
**Pull request by thegands**

### CODE REVIEW

File: bin/tictactoe [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+if players == "0"
+  Game.new(Player::Computer.new(xfirst == "1" ? "X" : "O"), Player::Computer.new(xfirst == "1" ? "O" : "X")).play
+elsif players == "1"
+  puts "Would you like to make the first or second move? (1/2)"
+  first_move = gets.chomp
+  first_move = gets.chomp until first_move.to_i.between?(1, 2)
+  first_move == "1" ? Game.new(Player::Human.new(xfirst == "1" ? "X" : "O"), Player::Computer.new(xfirst == "1" ? "O" : "X")).play : Game.new(Player::Computer.new(xfirst == "1" ? "X" : "O"), Player::Human.new(xfirst == "1" ? "O" : "X")).play
+else
+  Game.new(Player::Human.new(xfirst == "1" ? "X" : "O"), Player::Human.new(xfirst == "1" ? "O" : "X")).play
+end
```
Comment (randallreedjr) [inline bin/tictactoe]: Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  def valid_move?(input)
+    input.to_i.between?(1, 9) && !taken?(input)
+  end
+
+  def update(input, player)
+    cells[input.to_i-1] = player.token
+  end
+
+end
```
Comment (randallreedjr) [inline lib/board.rb]: Really good job reusing methods throughout this class (`reset!`,`turn_count`,`taken`)

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  def initialize(p1 = Player::Human.new("X"), p2 = Player::Human.new("O"), board = Board.new)
+    @player_1, @player_2, @board = p1, p2, board
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw? || full?
```
Comment (randallreedjr) [inline lib/game.rb]: Can the board ever be `full?` without someone having `won?` or the game being a `draw?`?

File: lib/players/computer.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  def center(board)
+    "5" unless board.taken?(5)
+  end
+
+  def corner(board)
+    corners = ["1", "3", "7", "9"]
+    if corners.map { |corner_value| board.taken?(corner_value) }.include?(false)
+      play = corners.sample
+      play = corners.sample while board.taken?(play)
+      play
```
Comment (randallreedjr) [inline lib/players/computer.rb]: I wonder if all three of these lines are necessary...

**Comment by randallreedjr** [comment]
@thegands Great job! I left you a few comments, feel free to respond. Closing this PR.

**Comment by thegands** [comment]
thank you for the feedback! i have refactored the code :smile:

**Pull request closed without merging**